### PR TITLE
WT-6538 Fix onpage prepare visibility check if the start and stop are from the same prepared transaction

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: test/suite/test_prepare14.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+import wttest
+from wiredtiger import WT_NOTFOUND
+from wtscenario import make_scenarios
+
+def timestamp_str(t):
+    return '%x' % t
+
+# test_prepare14.py
+# Test that visiblity check is OK when both the stop time
```
Comment (kommiharibabu) [inline test/suite/test_prepare14.py:37]: `Test that the transaction visibility of an on-disk update that has both the start and the stop time points are from the same uncommitted prepared transaction.`?

File: test/suite/test_rollback_to_stable19.py [hunk 2]
```diff
@@ -102,11 +102,17 @@ def test_rollback_to_stable_no_history(self):
         # Search for the key so we position our cursor on the page that we want to evict.
         self.session.begin_transaction("ignore_prepare = true")
         evict_cursor.set_key(1)
-        evict_cursor.search()
+        self.assertEquals(evict_cursor.search(), WT_NOTFOUND)
         evict_cursor.reset()
         evict_cursor.close()
         self.session.commit_transaction()
 
+        self.session.begin_transaction("ignore_prepare = true")
```
Comment (kommiharibabu) [inline test/suite/test_rollback_to_stable19.py:110]: As you are already testing the on-disk visibility separately in the above test, this test change may not be needed. If you want to keep this change also, then add a similar case in the below flow also `test_rollback_to_stable_with_history`.

File: src/include/txn_inline.h [hunk 3]
```diff
@@ -631,7 +631,14 @@ __wt_txn_tw_stop_visible(WT_SESSION_IMPL *session, WT_TIME_WINDOW *tw)
 static inline bool
 __wt_txn_tw_start_visible(WT_SESSION_IMPL *session, WT_TIME_WINDOW *tw)
 {
-    return ((WT_TIME_WINDOW_HAS_STOP(tw) || !tw->prepare) &&
+    /*
+     * Check the prepared flag if there is no stop time point or the start and stop time points are
+     * from the same transaction.
+     */
+    return (((WT_TIME_WINDOW_HAS_STOP(tw) &&
```
Comment (kommiharibabu) [inline src/include/txn_inline.h:638]: Do a similar change in the function `__wt_txn_tw_start_visible_all` also.

File: test/suite/test_rollback_to_stable19.py (hunk 2 above)
Comment (quchenhao) [reply test/suite/test_rollback_to_stable19.py:110]: OK

File: src/include/txn_inline.h (hunk 3 above)
Comment (quchenhao) [reply src/include/txn_inline.h:638]: good point

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Pull request merged**